app/main/posts/views/post-view-data.directive.js

```diff
@@ -214,24 +212,26 @@ function PostViewDataController(
             }
         });
 
-        $scope.$watch(function () {
-            return $location.path();
-        }, function (newValue, oldValue) {
-            if ($scope.editMode.editing) {
-                var postId = newValue.match(/^\/posts\/([0-9]+)(\/|$)/);
-                var locationUrlMatch = $location.path().match(/^\/posts\/([0-9]+)(\/|$)/);
-                if (postId && postId.length > 1 && !locationUrlMatch) {
-                    var tmpPost = _.filter($scope.posts, function (postItm) {
-                        return postItm.id === parseInt(postId[1]);
-                    });
-                    if (tmpPost.length > 0) {
-                        $scope.selectedPost.post = tmpPost[0];
-                        $scope.selectedPostId = tmpPost[0].id;
-                        $scope.editMode.editing = false;
-                    }
-                }
-            }
-        });
+        // I don't know what this does but its terrifying
+        //
+        // $scope.$watch(function () {
```

Comment:
@willdoran or @rowasc if you can explain what this did it would really help. Then we can hopefully reproduce it with something less scary looking

Reply:
ehh. Robbie I dont know what branch you based this off, but that code was GONE , same as the editMode checks from below in my PR from a couple days ago were I completely replaced it.. we even discussed it.

Reply:
ok this is proof that I was both in need of coffee and that we have very similar looking weirdness in both data and data editor, because I replaced something similar a couple days ago which is what we actually discussed.  The stuff I found and fixed was the data editor version of this , which was just as wrong and scary. 
Let me take a look today,we shouldn't need it anymore . This was used to manage the selected post as the URL changed. I think we don't need to replace it at all, because it's done in the route state management  but I'll check and report back. 
I'm really sorry about the confusion :(